### PR TITLE
Add a dependency management checklist

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -69,6 +69,26 @@ guide](https://github.com/agis/git-style-guide).
   _how_ it addresses the underlying issue, and what _side-effects_ it might
   have.
 
+## Dependency Management
+
+When introducing a new or updating an existing dependency, please consider the
+following tasks:
```
Comment (mavam) [inline contributing.md]:
```suggestion
When introducing a new dependency or updating an existing one, please consider the
following tasks:
```

File: contributing.md [hunk 2]
```diff
@@ -69,6 +69,26 @@ guide](https://github.com/agis/git-style-guide).
...
+
+- **CMake:** Reason whether the dependency is a public, private, or interface
```
Comment (mavam) [inline contributing.md]:
```suggestion
- **CMake:** Determine whether the dependency is a public, private, or interface
```

File: contributing.md [hunk 3]
```diff
@@ -69,6 +69,26 @@ guide](https://github.com/agis/git-style-guide).
...
+- **Nix**: Update the Nix environment to link against the new depdendency.
+
+- **Dockerfile**: Update `Dockerfile` and `Dockerfile_prebuilt` to install new
+  dependencies.
+
+- **CI**: Install new dependencies in GitHub Actions workflow files, including
+  the Jupyter Notebook workflow.
+
+- **Homebrew**: Notify the Homebrew package maintainer about the new
+  dependency.
```
Comment (mavam) [inline contributing.md]:
```suggestion
- **Homebrew**: Notify the Homebrew package maintainer about the dependency
  change.
```

File: contributing.md [hunk 4]
```diff
@@ -69,6 +69,26 @@ guide](https://github.com/agis/git-style-guide).
...
 
+## Dependency Management
+
+When introducing a new or updating an existing dependency, please consider the
+following tasks:
+
+- **CMake:** Reason whether the dependency is a public, private, or interface
+  depdency or existing targets.
+
+- **Nix**: Update the Nix environment to link against the new depdendency.
```
Comment (mavam) [inline contributing.md]:
```suggestion
- **Nix**: Update the Nix environment to link against the proper dependency.
```

File: contributing.md [hunk 5]
```diff
@@ -69,6 +69,26 @@ guide](https://github.com/agis/git-style-guide).
...
+
+- **Dockerfile**: Update `Dockerfile` and `Dockerfile_prebuilt` to install new
+  dependencies.
```
Comment (mavam) [inline contributing.md]:
```suggestion
- **Dockerfile**: Update `Dockerfile` and `Dockerfile_prebuilt` to reflect the
  dependency changes.
```

File: contributing.md [hunk 6]
```diff
@@ -69,6 +69,26 @@ guide](https://github.com/agis/git-style-guide).
...
+
+- **CI**: Install new dependencies in GitHub Actions workflow files, including
```
Comment (mavam) [inline contributing.md]:
```suggestion
- **CI**: Update the dependencies in GitHub Actions workflow files, including
```

File: contributing.md [hunk 7]
```diff
@@ -69,6 +69,26 @@ guide](https://github.com/agis/git-style-guide).
...
+  the Jupyter Notebook workflow.
+
+- **Homebrew**: Notify the Homebrew package maintainer about the new
+  dependency.
+
+- **Debian**: Notify the Debian package maintainer about the new dependency.
```
Comment (mavam) [inline contributing.md]:
```suggestion
- **Debian**: Notify the Debian package maintainer about the dependency changes.
```

**Review by mavam — CHANGES_REQUESTED** [review]
Good content, just wording tweaks.

**Review by mavam — CHANGES_REQUESTED** [review]
Yikes, how did the HTML sneak in there?

**Comment by dominiklohmann** [comment]
> Yikes, how did the HTML sneak in there?

Somewhere in the "Add suggestions to batch" mode. Thanks for noticing, will fix manually rn.

File: contributing.md [hunk 8]
```diff
@@ -621,26 +642,7 @@ When integrating 3rd-party code into the code base, use the following scaffold:
...
-  output of failed configurations.
-
-- *integration*: this checks whether the [integration tests][integration] pass,
-  similar to the *tests* check.
-
-[integration]: https://github.com/vast-io/vast/tree/master/integration
+We use GitHub Actions to build and test each commit. Merging a pull request
+requires that all checks pass for the latest commit in the branch. GitHub
+displays the status of the individual checks in the pull request. Passing all
+checks is mandatory for every pull request.
```
Comment (0snap) [inline contributing.md]: There is a slight redundancy here.

> Merging a pull request requires that all checks pass for the latest commit in the branch
> Passing all checks is mandatory for every pull request.

**Review by mavam — APPROVED** [review]

**Pull request merged**